### PR TITLE
[Gh 1301] Enhancement Feature - Bulk share reapply on dataset 

### PR DESCRIPTION
**Pull request by TejasRGitHub**
### Feature or Bugfix
- Feature

### Detail

- Adds feature to reapply shares in bulk for a dataset. 
- Also contains bugfix for AWS worker lambda errors 

### Relates
- https://github.com/data-dot-all/dataall/issues/1301
- https://github.com/data-dot-all/dataall/issues/1364

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)? N/A
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization? N/A
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features? N/A
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users? N/A
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by noah-paige** [comment]
@TejasRGitHub - looks like there is a merge conflict that needs to be addressed

Can you merge with latest in `main`?

**Comment by TejasRGitHub** [comment]
HI @noah-paige , Merged  with main and updated the PR

File: backend/dataall/modules/s3_datasets_shares/api/mutations.py [hunk 1]
```diff
@@ -10,3 +11,10 @@
     type=gql.Boolean,
     resolver=verify_dataset_share_objects,
 )
+
+reApplyShareObjectItemsOnDataset = gql.MutationField(
+    name='reApplyShareObjectItemsOnDataset',
+    args=[gql.Argument(name='datasetUri', type=gql.String)],
```
Comment (noah-paige) [inline backend/dataall/modules/s3_datasets_shares/api/mutations.py:17]: make nonnull string maybe to enforce value present

File: frontend/src/modules/Shares/services/reApplyShareObjectItemsOnDataset.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { gql } from 'apollo-boost';
+
+export const reApplyShareObjectItemsOnDataset = ({ datasetUri }) => ({
+  variables: {
+    datasetUri
+  },
+  mutation: gql`
+    mutation reApplyShareObjectItemsOnDataset($datasetUri: String) {
```
Comment (noah-paige) [inline frontend/src/modules/Shares/services/reApplyShareObjectItemsOnDataset.js:8]: if make nonnull string on backend need to change to `String!` here

File: backend/dataall/modules/shares_base/services/share_item_service.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import logging
 
 from dataall.core.permissions.services.resource_policy_service import ResourcePolicyService
+from dataall.core.stacks.aws.ecs import Ecs
```
Comment (noah-paige) [inline backend/dataall/modules/shares_base/services/share_item_service.py:4]: any need for this import?

File: backend/dataall/modules/shares_base/tasks/share_reapplier_task.py [hunk 4]
```diff
@@ -18,32 +18,68 @@
...
+                    session=session,
+                    share_uri=share_object.shareUri,
+                    old_status=ShareItemHealthStatus.Unhealthy.value,
+                    new_status=ShareItemHealthStatus.PendingReApply.value,
+                )
+                SharingService.reapply_share(engine, share_uri=share_object.shareUri)
+            return processed_share_objects
+
+    @classmethod
+    def process_repply_shares(cls, engine):
```
Comment (noah-paige) [inline backend/dataall/modules/shares_base/tasks/share_reapplier_task.py:46]: typo `process_reapply_shares`

File: backend/dataall/modules/shares_base/tasks/share_reapplier_task.py [hunk 5]
```diff
@@ -18,32 +18,68 @@
...
-                share_uri=share_object.shareUri,
-                old_status=ShareItemHealthStatus.Unhealthy.value,
-                new_status=ShareItemHealthStatus.PendingReApply.value,
-            )
-            SharingService.reapply_share(engine, share_uri=share_object.shareUri)
-        return processed_share_objects
+    if dataset_uri:
+        return EcsBulkShareRepplyService.process_reapply_shares_for_dataset(engine, dataset_uri)
+    else:
+        return EcsBulkShareRepplyService.process_repply_shares(engine)
```
Comment (noah-paige) [inline backend/dataall/modules/shares_base/tasks/share_reapplier_task.py:76]: typo `process_reapply_shares`

File: backend/dataall/modules/shares_base/services/share_item_service.py (hunk 3 above)
Comment (TejasRGitHub) [reply backend/dataall/modules/shares_base/services/share_item_service.py:4]: Removing this and few more imports which are not used

**Comment by TejasRGitHub** [comment]
Hi @noah-paige , updated the code with requested changes

**Review by noah-paige — APPROVED** [review]
looks good and tested locally - approving!

**Pull request merged**